akanda/rug/state.py

```diff
@@ -237,7 +237,9 @@ def transition(self, action, worker_context):
             return StopVM(self.params)
         elif self.vm.state == vm_manager.ERROR:
             return CalcAction(self.params)
-        return CheckBoot(self.params)
+        elif self.vm._currently_booting:
```

Comment:
This is using a private variable. Which I've tentatively removed in the version for Kilo.  Seems like we're not setting the vm.state = BOOTING in the boot() method. If we did that you could check for vm.state == BOOTING here and maintain the public interface.